src/doc/rustc/src/platform-support/fuchsia.md

```diff
@@ -0,0 +1,280 @@
+```
+
+`SDK_PATH` must be an environment variable containing the path to the downloaded
+Fuchsia SDK. These can be run together in a shell environment by executing
+`(source config-env.sh && ./x.py install)`.
+
+## Building Rust programs
+
+After compiling Rust binaries, you'll need to build a component, package it, and
+serve it to a Fuchsia device or emulator. All of this can be done using the
+Fuchsia SDK.
+
+As an example, we'll compile and run this simple program on a Fuchsia emulator:
+
+**`hello_fuchsia.rs`**
+```rust
+fn main() {
+    println!("Hello Fuchsia!");
+}
+
+#[test]
+fn it_works() {
+    assert_eq!(2 + 2, 4);
+}
+```
+
+Create a new file named `hello_fuchsia.rs` and fill out its contents with that
+code.
+
+### Create a package
+
+On Fuchsia, a package is the unit of distribution for software. We'll need to
+create a new package directory where we will place files like our finished
+binary and any data it may need. The working directory will have this layout:
+
+```
+hello_fuchsia.rs
+hello_fuchsia.cml
+package
+┣━ bin
+┃  ┗━ hello_fuchsia
+┣━ meta
+┃  ┣━ package
+┃  ┗━ hello_fuchsia.cm
+┗━ hello_fuchsia.manifest
+```
+
+Make the `package`, `package/bin`, and `package/meta` directories and create the
+following files inside:
+
+**`package/meta/package`**
+```json
+{"name":"hello_fuchsia","version":0}
+```
+
+The `package` file describes our package's name and version number. Every
+package must contain one.
+
+**`package/hello_fuchsia.manifest`**
+```
+bin/hello_fuchsia=${WORK_DIR}/bin/hello_fuchsia
+lib/ld.so.1=${SDK_PATH}/arch/${ARCH}/sysroot/dist/lib/ld.so.1
+lib/libfdio.so=${SDK_PATH}/arch/${ARCH}/lib/libfdio.so
+meta/package=${WORK_DIR}/meta/package
+meta/hello_fuchsia.cm=${WORK_DIR}/meta/hello_fuchsia.cm
+```
+
+Where `${WORK_DIR}` is the **absolute** path to the root of your directory.
+
+The `.manifest` file will be used to describe the contents of the package by
+relating their location when installed to their location on the file system. You
+can use this to make a package pull files from other places, but for this
+example we'll just be placing everything in the `package` directory.
+
+### Compiling a binary
+
+Using your freshly compiled `rustc`, you can compile a binary for Fuchsia using
+the following options:
+
+* `--target x86_64-fuchsia`/`--target aarch64-fuchsia`: Targets the Fuchsia
+  platform of your choice
+* `-L ${SDK_PATH}/arch/${ARCH}/lib`: Link against Fuchsia libraries from the SDK
+* `-L ${SDK_PATH}/arch/${ARCH}/sysroot/lib`: Link against Fuchsia kernel
+  libraries from the SDK
+
+Putting it all together:
+
+```sh
+# Configure these for the Fuchsia target of your choice
+# TARGET_ARCH={x86_64-fuchsia|aarch64-fuchsia}
+# ARCH={x64|aarch64}
+
+rustc --target ${TARGET_ARCH} -L${SDK_PATH}/arch/${ARCH}/lib -L${SDK_PATH}/arch/${ARCH}/sysroot/lib -o package/bin/hello_fuchsia hello_fuchsia.rs
+```
+
+### Bulding a component
+
+On Fuchsia, components require a component manifest written in CML. Here's a
```

Comment:
This prompts the question "what's CML", you can link to https://fuchsia.dev/fuchsia-src/concepts/components/v2/component_manifests or https://fuchsia.dev/reference/cml